presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedPlanStatisticsCalculator.java

```diff
@@ -93,7 +97,11 @@ public void registerPlan(PlanNode root, Session session)
             }
         });
         try {
-            historyBasedStatisticsCacheManager.getStatisticsCache(session.getQueryId(), historyBasedPlanStatisticsProvider).getAll(planNodesWithHash.build());
+            historyBasedStatisticsCacheManager.getStatisticsCache(session.getQueryId(), historyBasedPlanStatisticsProvider, getHistoryBasedOptimizerTimeoutLimit(session)).getAll(planNodesWithHash.build());
+            if (historyBasedStatisticsCacheManager.loadHistoryFailed(session.getQueryId())) {
+                session.getWarningCollector().add((new PrestoWarning(QUERY_OPTIMIZER_FAILURE, format("'%s' failed to load history query statistics", HistoryBasedPlanStatisticsCalculator.class.getSimpleName()))));
+                session.getOptimizerInformationCollector().addInformation(new PlanOptimizerInformation(HistoryBasedPlanStatisticsCalculator.class.getSimpleName(), false, Optional.empty(), Optional.of(true)));
```

Comment:
When HBO register function failed to load statistics, add warning and also add to logging